app/schema/answers/integer_answer.py

```diff
@@ -1,21 +1,24 @@
 from app.schema.answer import Answer
 from app.validation.positive_integer_type_check import PositiveIntegerTypeCheck
 from app.validation.integer_type_check import IntegerTypeCheck
+from app.schema.widgets.text_widget import TextWidget
 
 
 class IntegerAnswer(Answer):
-    def __init__(self):
-        super().__init__()
+    def __init__(self, answer_id=None):
+        super().__init__(answer_id)
         self.type_checkers.append(IntegerTypeCheck())
+        self.widget = TextWidget(self.id)
 
     def _cast_user_input(self, user_input):
         return int(user_input)
 
 
 class PositiveIntegerAnswer(Answer):
```

Comment:
Why does this not subclass from IntegerAnswer? The TypeCheck could then make use of mixins to check for positive number and IntegerChecks. Not that big a deal here but as we get to more complex types would reduce boilerplate.